org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/environments/ExecutionEnvironmentDescription.java

```diff
@@ -301,26 +308,22 @@ public LibraryLocation[] getLibraryLocations() {
 	 */
 	public String getVMArguments() {
 		StringBuilder arguments = new StringBuilder();
-		Iterator<Entry<String, String>> entries = fProperties.entrySet().iterator();
-		while (entries.hasNext()) {
-			Entry<String, String> entry = entries.next();
-			String key = entry.getKey();
-			String value = entry.getValue();
+		fProperties.forEach((key, value) -> {
 			boolean appendArgument = !key.startsWith(EE_ARG_FILTER);
 			if (appendArgument) {
 				arguments.append(key);
 				if (!value.isEmpty()) {
 					arguments.append('=');
 					value = resolveHome(value);
```

Comment:
why is resolveHome called again?

Reply:
Good question.
Propably because one gets the original and modifiable fProperties Map via the `getProperties()` method and can therefore alter the map's content.
I have added another commit to replace all the usages of that method and depreacte `getProperties()` for removal.
That change also replaced all usages of that method in the JDT code-base and AFAICT there is no other use in the Eclipse Platform.

I assume JDT follows the usual two (or was it three years?) deprecation policy before it can be removed? Or do you think that method should be kept?